[Docs] Add common mistakes page

AugustinLF opened the pull request:
I observed that newcomers (based on this repo issues and difficulties that people in my team have) tend to encounter the same 4/5 issue. Perhaps a "common mistakes", troubleshooting or something like that could be helpful. So here is a simple proposition :)

I divided this section in two parts, firstly, the problems that people have by not understanding enough Flow's type system, and secondly, flow current limitations.

I struggle to understand how the website is built, so I just added the page, feel free to show me the right direction to make this change work!

Lastly, I saw that there is a "tips" folder in the docs, perhaps this page would better belong there, however, I don't think it's published, so I preferred to add it to the more common `lang` folder.

mrkev commented:
IMO this is a great idea, sad it didn't get more eyes before. I'll try to import once the CI lets me

AugustinLF commented:
Thanks @mrkev, I do think that a lot of PR for doc improvements flight under the radar, which is quite sad, the community being pretty motivated on that!

Regarding this PR, be wary, I created a page, without importing it in anywhere (or perhaps it's automatic?), so you might have to do it yourself, or point me to the right direction!

mrkev commented:
I'll be completely honest, I'm well acquainted with how the website works. I think it's automatic though?

AugustinLF commented:
Hey @mrkev, is there anything you'd like me to do to make this happen? :)

jbrown215 commented on website/en/docs/lang/common-mistakes.md:
```diff
@@ -0,0 +1,76 @@
+
+### Common user errors <a class="toc" id="toc-common-user-errors" href="#toc-common-user-errors"></a>
+
+#### Refinement invalidation
```
I checked that `foo.bar` is not `null`, but Flow still thinks it is. Why does this happen?

jbrown215 commented on website/en/docs/lang/common-mistakes.md:
```diff
@@ -0,0 +1,76 @@
+
+#### Refinement invalidation
+
+You have to keep in mind in JavaScript, you can modify data structures as you want, and Flow takes that in consideration. As of today, Flow cannot identify side-effects, so you end up with Flow considering that a `console.log()` can modify the content of an array (since nothing prevents you from modifying `console.log`). Therefore, whenever you refine types, some of your assumptions might not be  ...
```
Can we add a basic refinement invalidation example here, too?

jbrown215 commented on website/en/docs/lang/common-mistakes.md:
````diff
@@ -0,0 +1,76 @@
+Another difficulty people encounter is when they want to use a type different than the one declared that they believe compatible (as a parameter to a function for instance), and Flow complains, telling that the two types are not compatible.
+
+```js
+type WithError = {
````
This example passes Flow :)

AugustinLF replied:
Good catch (I believe it used to fail)!

jbrown215 commented on website/en/docs/lang/common-mistakes.md:
````diff
@@ -0,0 +1,76 @@
+handleAnswer(res); // Flow is happy!
+```
+
+### Errors due to Flow limitations <a class="toc" id="toc-flow-errors" href="#toc-flow-errors"></a>
````
Can we change this section to "Why do my refinements disappear inside a closure?"

AugustinLF replied:
Hmmm, I'm not sure it's the same thing. Here, it's more about "you're right, but it doesn't work with flow implementation". I could add an example with closures (which are indeed another common problem).

jbrown215 replied:
For sure, I just would prefer to get the heading down to a specific question if you can. There are lots of Flow limitations and this example doesn't cover all of them :)

AugustinLF commented:
I pushed again! Sorry, git didn't get it when I renamed the file :'( I added examples about closures, and switched all the titles to questions. It should be better, and more beginner friendly!

jbrown215 commented:
Thanks for the PR! And sorry for the delay in my review!

jbrown215 commented:
I'm going to land this diff tomorrow! I expect to make minimal edits, but I like what you have here!

jbrown215 commented:
@AugustinLF: I ended up making it into it's own guide, so you can find it at flow.org/en/docs/ and click FAQ!

AugustinLF commented:
Thanks a lot, pretty excited that we got this merged 😃
